test/ddt_suite/record_union.lua

```diff
@@ -15,6 +15,17 @@ t {
     output = '["Hello, world!", 0, null, 42]'
 }
 
+--[[
+-- gh-64: treat lack of value (B) as null;
+-- validate was fixed, but flatten didn't
+t {
+    schema = foo,
+    func = 'flatten',
+    input = '{"A":"Hello, world!","C":42}',
+    output = '["Hello, world!", 0, null, 42]'
+}
+]]--
```

Comment:
I prefer not add this comment, because 
1. the person who will implement this feature can forget about this comment and add his own test
2. It is very easy to write the test

Reply:
I would prefer to leave it here as some kind of problem statement for a developer.

Alternatively I can close gh-64 here and file separate issue re flatten.

Reply:
I do not agree, but I do not insist. It is ok.

Reply:
Do you not agree with which of the two statements above?

Reply:
I do not agree that this comment is suitable for the patch.
However, in case you insist let it be here instead of closing #64

Reply:
Already marked #64 for closing and opened #67 as follow up.